doctr/utils/metrics.py

```diff
@@ -142,6 +144,32 @@ def box_iou(boxes_1: np.ndarray, boxes_2: np.ndarray) -> np.ndarray:
     return iou_mat
 
 
+def rbox_iou(boxes_1: np.ndarray, boxes_2: np.ndarray) -> np.ndarray:
+    """Compute the IoU between two sets of bounding boxes
+
+    Args:
+        boxes_1: bounding boxes of shape (N, 5) in format (x, y, w, h, alpha)
+        boxes_2: bounding boxes of shape (M, 5) in format (x, y, w, h, alpha)
+    Returns:
+        the IoU matrix of shape (N, M)
+    """
+
+    iou_mat = np.zeros((boxes_1.shape[0], boxes_2.shape[0]), dtype=np.float32)
+
+    if boxes_1.shape[0] > 0 and boxes_2.shape[0] > 0:
+        _pts1 = [cv2.boxPoints(((x, y), (w, h), alpha)) for [x, y, w, h, alpha] in list(boxes_1)]
+        _pts2 = [cv2.boxPoints(((x, y), (w, h), alpha)) for [x, y, w, h, alpha] in list(boxes_2)]
+        pts1 = [[(x, y) for [x, y] in list(pt)] for pt in _pts1]
+        pts2 = [[(x, y) for [x, y] in list(pt)] for pt in _pts2]
+        polys1 = [Polygon(pt) for pt in pts1]
+        polys2 = [Polygon(pt) for pt in pts2]
+        for i in range(boxes_1.shape[0]):
+            for j in range(boxes_2.shape[0]):
+                iou_mat[i, j] = polys1[i].intersection(polys2[j]).area / polys1[i].union(polys2[j]).area
+
+    return iou_mat
```

Comment:
Improvement suggestion : for IoU with polygons and boxes that aren't straight,
- convert them to boolean 2D maps
- perform `np.logical_and(...).sum() / np.logical_or(...).sum()` to get the IoU

that way it will be faster (bool instead of floats) and have fewer dependencies